en/lessons/gravity-model.md

```diff
@@ -0,0 +1,712 @@
+
+```
+install.packages("MASS")
+library(MASS)
+
+
+gravityModelData <- read.csv("VagrantsExampleData.csv")
+
+gravityModel <- glm.nb(vagrants~log(population)+log(distance)+wheat+wages+wageTrajectory, data=gravityModelData)
+summary(gravityModel)
+```
+
+Copy the above code into your *weightingsCalculations.r* file and save. You can now run the code using your favourite *R* environment (I use [RStudio](https://www.rstudio.com/)) and the results of the calculation should appear in the console window (what this looks like will depend upon your environment). You may need to set the [Working Directory](https://en.wikipedia.org/wiki/Working_directory) of your *R* environment to the directory containing your .csv and .r files. If you are using RStudio you can do this via the menus (Session -> Set Working Directory -> Choose Directory). You can also achieve the same with the command:
+
+```
+setwd(PATH) #change "PATH" to the full location on your computer where the files can be found  
+
+```
+
+Notice that line 4 is the line that solves the equation for us, using the [glm.nb](https://stat.ethz.ch/R-manual/R-devel/library/MASS/html/glm.nb.html) function, which is short for "generalized linear model - negative binomial". This line requires a number of inputs:
+
+* our variables using the column headers as written in the .csv file, along with any logging that must be done to them (vagrants, log(population), log(distance), wheat, wages, wageTrajectory). If you were running a model with your own data, you would adjust these to reflect your column headers in your dataset.
+* where the code can find the data - in this case a variable we've defined in line 3 called "gravityModelData".
+* and instructions on what to do if a value is missing (na.action=na.exclude).
+
+The outputs of the calculation can be seen in Figure 9:
+
+{% include figure.html filename="figure9.png" caption="Figure 9: The summary of the above code, showing the weightings for each variable and the y-intercept, listed under the 'Estimate' heading ($$\beta_{0}$$ to $$\beta_{5}$$. This summary also shows a number of other calculations, including [statistical significance](https://en.wikipedia.org/wiki/Statistical_significance)." %}
+
+## Step 3: Calculating the Estimates for each County
+
+Because we now have the y-intercept ($$β0$$), the weightings ($$β1-5$$), and the 5 variable values for each county ($$P$$, $$d$$, $$Wh$$, $$Wa$$, $$WaT$$), we have all the numbers we need to solve for the model's predicted value for a county: the final result.
+
+We have to do this once for each of the 32 counties.
+
+You could do this with a scientific calculator, by creating a spreadsheet formula, or writing a computer programme. To do this automatically in *R*, you can add the following to your code and re-run the programme. This *for* loop calculates the expected number of vagrants from each of the 32 counties in the example and prints the results for you to see:
```

Comment:
functions, variables, etc. used in _code_ should be enclosed in backticks so they render like `for`. This is extra important, because it will distinguish when you're referring to a term from the R code vs. the LaTeX that poitns to terms in the equations.

Reply:
done.